src/api/l_physics_world.c

```diff
@@ -110,6 +110,44 @@ static int l_lovrWorldNewSphereCollider(lua_State* L) {
   return 1;
 }
 
+static int l_lovrWorldNewMeshCollider(lua_State* L) {
+  World* world = luax_checktype(L, 1, World);
+  lovrAssert(lua_istable(L, 2), "Vertices must be a table");
+  lovrAssert(lua_istable(L, 3), "Indices must be a table");
+  int vertexCount = luax_len(L, 2);
+  int indexCount  = luax_len(L, 3);
+  // TODO: this never gets deallocated
+  float    * vertices = malloc(sizeof(float) * vertexCount * 3);
+  unsigned * indices  = malloc(sizeof(unsigned) * indexCount);
+
+  for (int i = 0; i < vertexCount; i++) {
+    lua_rawgeti(L, 2, i + 1);
+    lovrAssert(lua_istable(L, -1), "Each verticle must be a table of coordinates");
+    lua_rawgeti(L, -1, 1); // x
+    vertices[i * 3 + 0] = luaL_optnumber(L, -1, 0.);
+    lua_pop(L, 1);
+    lua_rawgeti(L, -1, 2); // y
+    vertices[i * 3 + 1] = luaL_optnumber(L, -1, 0.);
+    lua_pop(L, 1);
+    lua_rawgeti(L, -1, 3); // z
+    vertices[i * 3 + 2] = luaL_optnumber(L, -1, 0.);
+    lua_pop(L, 2);
+  }
+  for (int i = 0; i < indexCount; i++) {
+    lua_rawgeti(L, 3, i + 1);
+    indices[i] = luaL_checkinteger(L, -1) - 1;
+  }
+  lua_pop(L, indexCount);
```

Comment:
I wonder if there is a risk of overflowing the Lua stack by saving the pops until the end.

Reply:
I haven't thought of that. Strange the overflow never happened and I was testing with quite large trimeshes (icosphere with 5 subdivisions). It's an easy fix, I'll make the change.